packages/vscode-extension/src/commands/SetupAccount.ts

```diff
@@ -279,45 +273,53 @@ export default class SetupAccount extends BaseAction {
 			return;
 		}
 
-		const tokenId = await this.getTokenId();
-		if (!tokenId) {
+		const certificateId = await this.getCertificateId();
+		if (!certificateId) {
 			return;
 		}
 
-		const tokenSecret = await this.getTokenSecret();
-		if (!tokenSecret) {
+		const privateKeyFilePath = await this.getPrivateKeyFilePath();
+		if (!privateKeyFilePath) {
 			return;
 		}
 
-		const commandParams: { authid: string; account: string; tokenid: string; tokensecret: string; url?: string } = {
+		const commandParams: { authid: string; account: string; certificateid: string; privatekeypath: string; url?: string } = {
 			authid: authId,
 			account: accountId,
-			tokenid: tokenId,
-			tokensecret: tokenSecret,
+			certificateid: certificateId,
+			privatekeypath: privateKeyFilePath[0].fsPath,
 		};
 		if (url) {
 			commandParams.url = url;
 		}
 
-		const saveTokenPromise = AuthenticationUtils.saveToken(commandParams, getSdkPath(), this.rootWorkspaceFolder);
+		const authenticateCiPromise = AuthenticationUtils.authenticateCi(
+			commandParams,
+			getSdkPath(),
+			this.rootWorkspaceFolder,
+			new ExecutionEnvironmentContext({
+				platform: VSCODE_PLATFORM,
+				platformVersion: vscode.version,
+			}),
```

Comment:
Shouldn't we add this execEnv the browser based authentication?

Reply:
I think we should. It's looks like a bug from the original implementation.